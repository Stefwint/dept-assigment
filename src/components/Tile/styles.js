import styled from 'styled-components';
import { Box } from '../styles';
import { px2rem } from '../../utils';

const STile = styled(Box)`
  a {
    display: block;
    text-decoration: none;

    > div {
      overflow: hidden;
      transition: transform 0.3s ease;

      span {
        display: block;
        text-transform: uppercase;
      }

      img {
        display: block;
        width: 100%;
        max-height: ${px2rem(500)};
        object-fit: cover;
        transition: transform 0.3s ease;
      }
    }

    &:hover,
    &:focus {
      > div {
        transform: scale(0.95);

        img {
          transform: scale(1.08);
        }
      }
    }
  }
`;

export default STile;
