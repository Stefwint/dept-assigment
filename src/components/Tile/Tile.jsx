import React from 'react';
import PropTypes from 'prop-types';
import { useInView } from 'react-intersection-observer';

// Styling
import { ArrowButton, Box, Text } from '../styles';
import STile from './styles';
import theme from '../../theme';

const Tile = ({ title, client, image, link, isSmall }) => {
  const [ref, inView] = useInView({
    threshold: 0.2,
    triggerOnce: true,
  });

  return (
    <STile mb={isSmall ? [25, 0] : 0} pt={isSmall ? 20 : 0} ref={ref} inView={inView}>
      <a href={link} target="_blank">
        {image && !isSmall && (
          <Box mb={20}>
            <img src={`../../../static/images/${image}`} alt={title} />
          </Box>
        )}
        {client && (
          <Text as="span" fontSize={[14, 15]} fontWeight={700} color={theme.colors.gray}>
            {client}
          </Text>
        )}
        <Text
          as="h3"
          mt={15}
          mb={20}
          fontSize={!isSmall ? 30 : 24}
          fontWeight={400}
          color={theme.colors.black}
        >
          {title}
        </Text>
        <ArrowButton>View case</ArrowButton>
      </a>
    </STile>
  );
};

Tile.propTypes = {
  title: PropTypes.string.isRequired,
  client: PropTypes.string,
  image: PropTypes.string,
  link: PropTypes.string,
  isSmall: PropTypes.bool,
};

Tile.defaultProps = {
  client: '',
  image: '',
  link: '',
  isSmall: false,
};

export default Tile;
