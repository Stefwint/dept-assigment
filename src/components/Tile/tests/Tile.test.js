import React from 'react';
import renderer from 'react-test-renderer';
import Tile from '../Tile';

describe('testing Tile', () => {
  const item = (
    <Tile
      title="A summer island in the Netherlands"
      client="bol.com"
      link="https://www.deptagency.com"
      image="bolcom.png"
    />
  );

  it('Renders correctly', () => {
    const renderItem = renderer.create(item).toJSON();
    expect(renderItem).toMatchSnapshot();
  });
});
