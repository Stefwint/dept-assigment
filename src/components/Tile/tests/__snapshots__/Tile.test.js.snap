// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Tile Renders correctly 1`] = `
<div
  className="Flex__Box-ro60kk-0 styles__STile-sc-1rx2yf9-0 XMugF"
>
  <a
    href="https://www.deptagency.com"
    target="_blank"
  >
    <div
      className="Flex__Box-ro60kk-0 kflYyU"
    >
      <img
        alt="A summer island in the Netherlands"
        src="../../../static/images/bolcom.png"
      />
    </div>
    <span
      className="Flex__Text-ro60kk-1 hqPSXo"
      color="#939393"
      fontSize={
        Array [
          14,
          15,
        ]
      }
      fontWeight={700}
    >
      bol.com
    </span>
    <h3
      className="Flex__Text-ro60kk-1 irqMHd"
      color="#0E0E0E"
      fontSize={30}
      fontWeight={400}
    >
      A summer island in the Netherlands
    </h3>
    <span
      className="Button__ArrowButton-sc-1qg0mzm-1 iqQLXk"
    >
      View case
    </span>
  </a>
</div>
`;
